### PR TITLE
Add canal 3.28, deprecate 3.25

### PR DESCRIPTION
**Pull request by xrstf**
**What this PR does / why we need it**:
Regular maintenance. The release notes @ https://docs.tigera.io/calico/latest/release-notes/#important-upgrade-notes mention breaking changes that should not apply to us.

**What type of PR is this?**
/kind deprecation
/kind chore

**Does this PR introduce a user-facing change? Then add your Release Note here**:
```release-note
Add support for Canal 3.28, deprecate Canal 3.25.
```

**Documentation**:
```documentation
NONE
```

### CODE REVIEW

File: addons/canal/canal_v3.28.yaml [hunk 1]
```diff
@@ -0,0 +1,5214 @@
...
+      serviceAccountName: canal
+      # Minimize downtime during a rolling upgrade or deletion; tell Kubernetes to do a "force
+      # deletion": https://kubernetes.io/docs/concepts/workloads/pods/pod/#termination-of-pods.
+      terminationGracePeriodSeconds: 0
+      priorityClassName: system-node-critical
+      initContainers:
+        # This container installs the CNI binaries
+        # and CNI network config file on each node.
+        - name: install-cni
+          image: d'{{ Image "quay.io/calico/cni:v3.28.0" }}'
```
Comment (cnvergence) [inline addons/canal/canal_v3.28.yaml]:
```suggestion
          image: '{{ Image "quay.io/calico/cni:v3.28.0" }}'
```

**Review by cnvergence — CHANGES_REQUESTED** [review]

File: addons/canal/canal_v3.28.yaml (hunk 1 above)
Comment (xrstf) [reply addons/canal/canal_v3.28.yaml]: Aw good catch, I left that one in on purpose, to test you!

Comment (cnvergence) [reply addons/canal/canal_v3.28.yaml]: big secret

**Review by cnvergence — APPROVED** [review]
I think all CRDs are on board this time
/approve

**Pull request merged**